helix-core/src/comment.rs

```diff
@@ -292,37 +355,92 @@ pub fn create_block_comment_transaction(
             }
         }
     }
-    (Transaction::change(doc, changes.into_iter()), ranges)
+    (changes, ranges)
 }
 
 #[must_use]
 pub fn toggle_block_comments(
     doc: &Rope,
-    selection: &Selection,
+    ranges: &[Range],
     tokens: &[BlockCommentToken],
-) -> Transaction {
+    selections: &mut SmallVec<[Range; 1]>,
+    added_chars: &mut usize,
+    removed_chars: &mut usize,
```

Comment:
Why pass these in C++ style? Returning the ones added by the changes in a tuple seems more natural